File: app/src/main/java/org/ole/planet/myplanet/service/StayOnlineWorker.java
package org.ole.planet.myplanet.service;

import android.content.Context;
import android.content.Intent;

import androidx.annotation.NonNull;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import org.ole.planet.myplanet.utilities.Constants;
import org.ole.planet.myplanet.utilities.NetworkUtils;

public class StayOnlineWorker extends Worker {
    private Context context;

    public StayOnlineWorker(
            @NonNull Context context,
            @NonNull WorkerParameters workerParams
    ) {
        super(context, workerParams);
        this.context = context;
    }

    @NonNull
    @Override
    public Result doWork() {
        if (Constants.showBetaFeature(Constants.KEY_SYNC, context)) {
            if (NetworkUtils.isWifiConnected()) {
                LocalBroadcastManager.getInstance(context).sendBroadcast(new Intent("SHOW_WIFI_ALERT"));
            }
        }

        return Result.success();
    }
}

File: app/src/main/java/org/ole/planet/myplanet/service/StayOnlineWorker.kt
package org.ole.planet.myplanet.service

import android.content.Context
import android.content.Intent
import androidx.work.CoroutineWorker
import androidx.work.WorkerParameters
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import org.ole.planet.myplanet.utilities.Constants.isBetaWifiFeatureEnabled
import org.ole.planet.myplanet.utilities.NetworkUtils.isWifiConnected

class StayOnlineWorker(private val context: Context, workerParams: WorkerParameters) : CoroutineWorker(context, workerParams) {
    override suspend fun doWork(): Result {
        if (isBetaWifiFeatureEnabled(context)) {
            if (isWifiConnected()) {
                withContext(Dispatchers.IO) {
                    val broadcastService = getBroadcastService(context)
                    broadcastService.sendBroadcast(Intent("SHOW_WIFI_ALERT"))
                }
            }
        }
        return Result.success()
    }
}
